contracts/voting-verifier/src/execute.rs

```diff
@@ -274,7 +274,7 @@ fn create_messages_poll(
 ) -> Result<PollId, ContractError> {
     let id = POLL_ID.incr(store)?;
 
-    let poll = WeightedPoll::new(id, snapshot, block_height + expiry, poll_size);
+    let poll = WeightedPoll::new(id, snapshot, block_height.saturating_add(expiry), poll_size);
```

Comment:
same here

Reply:
I kind of feel the same way as here: https://github.com/axelarnetwork/axelar-amplifier/pull/296#discussion_r1525694763

Reply:
see my response there